src/openzaak/components/documenten/tests/management/test_delete_empty_eioc.py

```diff
@@ -0,0 +1,64 @@
+# SPDX-License-Identifier: EUPL-1.2
+# Copyright (C) 2024 Dimpact
+
+from django.core.management import call_command
+
+from rest_framework import status
+from rest_framework.test import APITestCase
+from vng_api_common.tests import reverse
+
+from openzaak.components.besluiten.models import BesluitInformatieObject
+from openzaak.components.besluiten.tests.factories import (
+    BesluitFactory,
+    BesluitInformatieObjectFactory,
+)
+from openzaak.components.zaken.models import ZaakInformatieObject
+from openzaak.components.zaken.tests.factories import (
+    ZaakFactory,
+    ZaakInformatieObjectFactory,
+)
+from openzaak.components.zaken.tests.utils import ZAAK_WRITE_KWARGS
+from openzaak.tests.utils import JWTAuthMixin
+
+from ...models import EnkelvoudigInformatieObjectCanonical
+from ..factories import EnkelvoudigInformatieObjectCanonicalFactory
+
+
+class DeleteEmptyEIOCTests(JWTAuthMixin, APITestCase):
+    heeft_alle_autorisaties = True
+
+    def test_basic(self):
+
+        eioc = EnkelvoudigInformatieObjectCanonicalFactory()
+        latest = eioc.latest_version
+        zaak = ZaakFactory()
+        ZaakInformatieObjectFactory(informatieobject=eioc, zaak=zaak)
+
+        besluit = BesluitFactory()
+        BesluitInformatieObjectFactory(informatieobject=eioc, besluit=besluit)
+
+        latest.delete()
```

Comment:
How about handling deleting? Should I prevent it if there is only one left? What if you want to delete a version? Should it only delete if the canonical is deleted?

Reply:
I'm not sure if it's possible via the API to delete a specific version, I would say that we should reflect the API behavior as much as possible, so you could check what happens now in these cases if you try to delete the EIO via the API

Reply:
@Coperh did you check the API behavior here?

Reply:
I agree with Steven, I think we can't delete the only document version without canonical object. 
You can modify this test by removing creating a document (and have something like `EnkelvoudigInformatieObjectCanonicalFactory(latest_version=None)`
I think this is what happened in the original issue. 

But it's optional, since the end result is the same

Reply:
Please ignore my previous comment, I forget that the validation was introduced to prevent it